app/services/datetime_parser.rb

```diff
@@ -6,34 +6,45 @@ def initialize(date:, time:)
     @raw_time = time.to_s
   end
 
+  def issues
+    @issue_items
+  end
+
+  def issues_for(param)
+    @issue_items.filter { |issue| issue.keys.first == param }.map { |issue| issue.values.first }
+  end
+
   def parse
-    begin
-      check_date_is_valid
-      check_time_is_valid
+    @issue_items = []
```

Comment:
This should be moved to the initialize method

Reply:
I thought so too, but if we call `parse` twice, we'd end up doubling the number of issues recorded. Issues should be cleared every time we run `parse`.

Reply:
Isn't a new instance of DateTimeParser called on each submit?

Reply:
It is, though in theory it doesn't have to be.

Have now moved this to the initialize method, tests continue to pass 👍